src/base/Roles/AccountantWithRateProviders.sol

```diff
@@ -294,6 +381,16 @@ contract AccountantWithRateProviders is Auth, IRateProvider {
 
         AccountantState storage state = accountantState;
         if (state.isPaused) revert AccountantWithRateProviders__Paused();
+
```

Comment:
I think we can create a function to avoid duplicating code:

```
if (vault.totalSupply() > 0 && lendingInfo.lendingRate > 0) {
    uint256 timeElapsed = block.timestamp - lendingInfo.lastAccrualTime;
    uint256 totalDeposits = vault.totalSupply().mulDivDown(accountantState.exchangeRate, ONE_SHARE);

    uint256 interestAccrued = totalDeposits.mulDivDown(
        lendingInfo.lendingRate * timeElapsed,
        SECONDS_PER_YEAR * BASIS_POINTS
    );

    uint256 protocolFees = totalDeposits.mulDivDown(
        lendingInfo.protocolFeeRate * timeElapsed,
        SECONDS_PER_YEAR * BASIS_POINTS
    );

    accountantState.feesOwedInBase += uint128(protocolFees);
    lendingInfo.lastAccrualTime = block.timestamp;
}
```